zeek/README.md

```diff
@@ -1,6 +1,412 @@
-This directory contains json type definitions for typed decoding of zeek ndjson logs.
+# JSON Type Definitions
 
-Currently it contains types.json, which was computed using the [print-types.zeek](https://github.com/brimsec/zeek/blob/master/brim/print-types.zeek) script running on Zeek 3.1.
+- [Summary](#summary)
+- [Contact us!](#contact-us)
+- [Sample data](#sample-data)
+- [Usage](#usage)
+- [Why is this even necessary?](#why-is-this-even-necessary)
+- [Type definition structure & importance of `_path`](#type-definition-structure--importance-of-_path)
+- [Customizing type definitions](#customizing-type-definitions)
+  * [Sample data](#sample-data-1)
+  * [Handling exceptions](#handling-exceptions)
+- [Older versions of Zeek/Bro](#older-versions-of-zeekbro)
+- [Need help? Have feedback?](#need-help-have-feedback)
 
-In the future it is expected to contain additional versions, generated on different versions/configurations of Zeek.
+# Summary
 
+The file [`types.json`](types.json) in this directory contains configuration
+that can be used to restore Zeek-compatible rich data types to events that
+have been output in JSON format by Zeek. The definitions in the current
+revision of `types.json` reflect the default set of logs output by
+[Zeek v3.1.2](https://github.com/zeek/zeek/releases/tag/v3.1.2) and are
+likely to be compatible with default logs from older Zeek/Bro releases as
+well.
+
+If your Zeek environment is customized, you will need to adjust the type
+definition to match the differences in your JSON logs. The sections below
+describe the customization process.
+
+# Contact us!
+
+If you're using JSON type defintions with `zq` or [Brim](https://github.com/brimsec/brim),
+we'd like to hear from you! We know the process of customizing the definitions
+can be tricky. We have ideas for ways we might improve it, but we'll have a
+better sense of priority and how to go about it if we've heard from those
+who have tried it. Whether you've used this feature and it "just worked" or if
+you hit challenges and need help, please join our
+[public Slack](https://join.slack.com/t/brimsec/shared_invite/zt-cy34xoxg-hZiTKUT~1KdGjlaBIuUUdg)
+and tell us about it. Thanks!
+
+# Sample data
+
+To see a working example before working with your own data, clone the
+[zq-sample-data](https://github.com/brimsec/zq-sample-data) repository.
+
+```
+# git clone --depth=1 https://github.com/brimsec/zq-sample-data ~/zq-sample-data
+```
+
+A set of JSON-format Zeek logs will now be present in `~/zq-sample-data/zeek-ndjson`.
+
+# Usage
+
+Asssuming you have the [`zq`](https://github.com/brimsec/zq) repository locally
```

Comment:
Should we briefly mention the existence of a json type inference mechanism, just so readers have a sense of what happens if you omit `-j` altogether?

Reply:
I added some stuff to mention inference in a section low down. Let me know what you think.